lib/system/iterators.nim

```diff
@@ -98,20 +98,23 @@ iterator mitems*(a: var cstring): var char {.inline.} =
     assert b == "aBc"
     assert a == "aBc\0def"
 
-  template impl() =
+  when nimvm:
     var i = 0
     let n = len(a)
     while i < n:
       yield a[i]
       inc(i)
-  when defined(js): impl()
   else:
-    when nimvm: impl()
+    when defined(js):
+      # xxx this could give a CT error in js RT, for example
+      # using similar approach as https://github.com/nim-lang/Nim/pull/15877,
+      # but note this isn't perfect as `compiles(block: someCstring[0] = 'x')` is true.
+      doAssert false, "js cstring cannot be mutated"
```

Comment:
We should really try `{.error: "jstring cannot be mutated".}`  here.

Reply:
that can't work:
```nim
proc main() =
  var a0 = "abc"
  var a = a0.cstring
  for ai in mitems(a):
    if ai == 'b': ai = 'B'
  echo a
# main() # good: gives CT error
static: main() # bad: gives CT error but should work
```
I have concrete ideas for how to fix that (and more generally, how to make `nimvm` more flexible), I think it's possible but not easy, and for now this PR improve things (avoids silently incorrect behavior)

Reply:
make sense